src/modules/services_manager_target.rb

```diff
@@ -139,8 +140,15 @@ def inspect
 
     def save
       return true unless modified?
+
       log.info('Saving default target...')
-      Yast2::Systemd::Target.set_default(default_target)
+      unless Yast2::Systemd::Target.find(self.default_target)
+        # TRANSLATORS: error popup, %s is the default target e.g. graphical
+         Report.Warning(_("Cannot find default target '%s' which is not available," \
```

Comment:
The warning sounds a bit weird to me. Why not something like:

```
Report.Warning(_("Target '%s' is not available. Using text mode as default target.") % default_target) 
```

Note that `self` is not required to get the `default_target` value.